packages/schematics/angular/service/index.ts

```diff
@@ -22,61 +20,61 @@ import {
   url,
 } from '@angular-devkit/schematics';
 import * as ts from 'typescript';
-import { addProviderToModule } from '../utility/ast-utils';
-import { InsertChange } from '../utility/change';
+import { getFirstNgModuleName } from '../utility/ast-utils';
 import { buildRelativePath, findModuleFromOptions } from '../utility/find-module';
 import { Schema as ServiceOptions } from './schema';
 
+function getModuleNameFromPath(host: Tree, modulePath: Path) {
+  if (!host.exists(modulePath)) {
+    throw new Error('Specified module does not exist');
```

Comment:
This should throw a `SchematicsException`

Reply:
Done.